fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -569,14 +690,33 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil)
       password_placeholder = @jwt.nil? ? 'YourPassword' : nil
       jwt_placeholder = @jwt.nil? ? nil : 'YourJWT'
 
-      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt)
-      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder))
+      # Handle AppStore Connect API
+      use_api_key = !@api_key.nil?
+      api_key_placeholder = nil unless use_api_key
+      api_key_placeholder = { key_id: "YourKeyID", issuer_id: "YourIssuerID" } if use_api_key
+
+      if use_api_key
+        # Specified p8 needs to be generated to call altool
+        tmp_p8_file_path = File.join(Dir.tmpdir, "AuthKey_#{@api_key[:key_id]}.p8")
+        File.open(tmp_p8_file_path, "w") do |p8|
+          key_content = @api_key[:is_key_content_base64] ? Base64.decode64(@api_key[:key]) : @api_key[:key]
+          p8.write(key_content)
+        end
+      end
+
+      command = @transporter_executor.build_upload_command(@user, @password, actual_dir, @provider_short_name, @jwt, platform, @api_key)
+      UI.verbose(@transporter_executor.build_upload_command(@user, password_placeholder, actual_dir, @provider_short_name, jwt_placeholder, platform, api_key_placeholder))
 
       begin
         result = @transporter_executor.execute(command, ItunesTransporter.hide_transporter_output?)
       rescue TransporterRequiresApplicationSpecificPasswordError => ex
         handle_two_step_failure(ex)
         return upload(app_id, dir, package_path: package_path, asset_path: asset_path)
+      ensure
+        if use_api_key
+          tmp_p8_file_path = File.join(Dir.tmpdir, "AuthKey_#{@api_key[:key_id]}.p8")
+          FileUtils.rm_f(tmp_p8_file_path)
+        end
```

Comment:
If you go with my suggestion above to use the traditional `Dir.mktmpdir` or `TempFile` to create the file, this would become just:

```suggestion
        FileUtils.rm_f(tmp_p8_file_path) unless tmp_p8_file_path.nil?
```